### ISSUE TITLE
Scheduler container crash when restarted after account creation

### ISSUE THREAD

**Issue by Mallear** [Netflix/security_monkey#885]
Hi,

We installed Security Monkey on an AWS instance with a RDS instance using the Docker quickstart and the provided compose file.

When adding a new active AWS account through the web UI and restarting the scheduler container, the container crash with the following stack trace :

```
2017-11-27 09:14:43,963 DEBUG: Scheduler adding account my_account [in /usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/scheduler.py:211]
DEBUG:security_monkey:Scheduler adding account my_account
2017-11-27 09:14:44,036 ERROR: Scheduler Exception: must be type, not None [in /usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/scheduler.py:233]
ERROR:security_monkey:Scheduler Exception: must be type, not None
ERROR:security_monkey:Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/scheduler.py", line 212, in setup_scheduler
    rep = Reporter(account=account)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/reporter.py", line 38, in __init__
    self.all_monitors = all_monitors(account, debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/monitors.py", line 76, in all_monitors
    monitor = Monitor(watcher_class, account, debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/monitors.py", line 19, in __init__
    self.watcher = watcher_class(accounts=[account.name], debug=debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/watchers/vpc/subnet.py", line 34, in __init__
    super(Subnet, self).__init__(accounts=accounts, debug=debug)
TypeError: must be type, not None

2017-11-27 09:14:44,036 ERROR: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/scheduler.py", line 212, in setup_scheduler
    rep = Reporter(account=account)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/reporter.py", line 38, in __init__
    self.all_monitors = all_monitors(account, debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/monitors.py", line 76, in all_monitors
    monitor = Monitor(watcher_class, account, debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/monitors.py", line 19, in __init__
    self.watcher = watcher_class(accounts=[account.name], debug=debug)
  File "/usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/watchers/vpc/subnet.py", line 34, in __init__
    super(Subnet, self).__init__(accounts=accounts, debug=debug)
TypeError: must be type, not None
 [in /usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/scheduler.py:234]
2017-11-27 09:14:44,036 DEBUG: Logging exception from scheduler with location: None to the database. [in /usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/datastore.py:726]
DEBUG:security_monkey:Logging exception from scheduler with location: None to the database.
2017-11-27 09:14:44,043 DEBUG: Completed logging exception to database. [in /usr/local/lib/python2.7/dist-packages/security_monkey-0.9.3-py2.7.egg/security_monkey/datastore.py:757]
DEBUG:security_monkey:Completed logging exception to database.
INFO:apscheduler.scheduler:Scheduler started
2017-11-27 09:14:44,043 INFO: Scheduler started [in /usr/local/lib/python2.7/dist-packages/APScheduler-2.1.2-py2.7.egg/apscheduler/scheduler.py:574]
2017-11-27 09:14:44,043 INFO: Scheduler has been shut down [in /usr/local/lib/python2.7/dist-packages/APScheduler-2.1.2-py2.7.egg/apscheduler/scheduler.py:606]
INFO:root:Shutting down thread pool
INFO:apscheduler.scheduler:Scheduler has been shut down
```

This behavior happens every time we set an active account.
Do you have a clue of what could be wrong ?

Thanks

**Comment by mstair** [comment]
Likely the same issue as https://github.com/Netflix/security_monkey/issues/862#issuecomment-343965895, just being observed with the watchers modules here (`subnet` has an OpenStack watcher defined).

I will rename all the OpenStack watchers to unique and submit a PR.

**Comment by mikegrima** [comment]
Just merged in #886 .  Can you check if that resolved your issue @Mallear ?

**Comment by Mallear** [comment]
I've just rebuild the container and test again: everything works well !

Thanks for your quick reply.
